lib/natalie/compiler/pass1.rb

```diff
@@ -139,10 +139,14 @@ def process_case(exp)
             (_, (_, *matchers), *when_body) = when_exp
             when_body = when_body.map { |w| process(w) }
             when_body = [s(:nil)] if when_body == [nil]
-            matchers.each do |matcher|
-              cond << s(:is_truthy, s(:public_send, process(matcher), s(:intern, '==='), s(:args, value_name), 'nullptr'))
-              cond << s(:block, *when_body)
-            end
+            full_condition = s(
+              :is_truthy,
+              # FIXME: Get rid of the :nil here
+              # FIXME: This is still a bit hacky
+              matchers.reduce(s(:nil)) {|new_cond, matcher| process_or(s(:or, new_cond, s(:public_send, process(matcher), s(:intern, '==='), s(:args, value_name), 'nullptr'))) }
```

Comment:
A way to represent a just boolean `or` might be nice here

Reply:
What's the motivation for this change? I think it's harder to read what's going on, but if there's a performance boost with it, I could see how it could be worth the change...

Reply:
Nevermind, I see in Discord you are fixing a bug where we compile lambdas inside the `when_body` twice. Carry on!

Reply:
Please add a test to prove this is fixed. I would hate to regress on that.

Reply:
I added a test for it. :hugs:

Reply:
Thanks!